tool/tsh/common/workload_identity.go

```diff
@@ -102,102 +103,102 @@ func (c *workloadIdentityIssueCommand) run(cf *CLIConf) error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
-
-	clusterClient, err := tc.ConnectToCluster(ctx)
-	if err != nil {
-		return trace.Wrap(err)
+	if cf.Headless {
+		tc.AllowHeadless = true
```

Comment:
From what I see, setting `tc.AllowHeadless = true` is more like saying "this command supports headless login".

Passing the `--headless` flag sets the connector to the headless connector:

https://github.com/gravitational/teleport/blob/0baf7a285e844df419e4683f009c8688dff1d594/tool/tsh/common/tsh.go#L3706-L3712

And then if the command invocation uses the headless connector, `getSSHLoginFunc` looks at `tc.AllowHeadless`:

https://github.com/gravitational/teleport/blob/0baf7a285e844df419e4683f009c8688dff1d594/lib/client/api.go#L3765-L3777

There are some places which just set `tc.AllowHeadless = true` and others that put it behind a conditional check on `cf.Headless`. I'm inclined to say that the check seems to be unnecessary. 🤔

Reply:
> I'm inclined to say that the check seems to be unnecessary. 🤔

You're probably right here - I looked at a few of the call sites and saw that those had the conditional and didn't want to get too deep into the internals of `tsh` (I rarely make `tsh` changes). Looking at those funcs you've linked, it's clearer it's not necessary. I'll remove the conditional.